src/lib/fcitx-utils/log.cpp

```diff
@@ -23,7 +23,10 @@ namespace {
 
 FCITX_DEFINE_LOG_CATEGORY(defaultCategory, "default");
 
+using log_rules_t = std::vector<std::pair<std::string, LogLevel>>;
+
 static std::ostream *defaultLogStream = &std::cerr;
```

Comment:
Since we have more than one global value, suggest to create a new struct to hold them all and have only one top level global variable.